src/pages/data/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { SetStateAction, useState } from "react";
+import { DmasAPI, DmasData, MockDmasAPI } from "../../api/dmas";
+import { useQuery } from "@tanstack/react-query";
+import { Label, BarChart,LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer, Bar, Rectangle } from 'recharts';
+import { forEachChild } from "typescript";
+
+
+// TODO replace this with a real implementation
+const Plants: DmasAPI = MockDmasAPI;
+
+function formatData(){
+    const dataArray = useQuery({
+        queryFn: Plants.getDmasData,
+        queryKey: ["species"],
+    });
+    const plantTotalsByYear = new Map<number, Record<string, number>>();
+    if (dataArray.data != undefined){
+        for(const x of dataArray.data){
+            for(const y of x.plantGrowth) {
+                const year = y.date.getFullYear();
+                if (!plantTotalsByYear.has(year)) {
+                    plantTotalsByYear.set(year, {});
+                }
+               plantTotalsByYear.get(year)![x.species] = 0;
+               plantTotalsByYear.get(year)![x.species] += y.count; 
+            } 
+        }
+    }
+    const plantTotalsArray = Array.from(plantTotalsByYear.entries())
+        .map(([year, totals]) => ({ year, ...totals }));
+    return plantTotalsArray   
+    }
+
+function returnAllSpecies() {
+    const allSpecies = [];
+    const returnData = useQuery({
+        queryFn: Plants.getDmasData,
+        queryKey: ["species"],
+      })
+      if(returnData.data != undefined){
+        for (const x of returnData.data) {
+            allSpecies.push(x.species)
+          }
+      }
+      return allSpecies;
+}
+function stringToColour(str: string){
```

Comment:
Am I right in guessing that this hashes the string, then turns it into a CSS hex code? If so could you please add a comment explaining?